spark-submit-flamegraph

```diff
@@ -183,8 +185,8 @@ password=profiler,prefix=sparkapp,tagMapping=spark"
   spark_cmd+=(--jars)
   spark_cmd+=("${jars}")
 
```

Comment:
Is the change spark_arg -> spark_cmd intentional?

Reply:
Yes. We use Apache Commands CLI for custom parameters so with `spark_args+=(--jars)`,
```bash
$ spark-submit-flamegraph --master "local[*]" --executor-memory 8g example-job-0.1.jar -i file:///tmp/input -o file:///tmp/output -d 2020-01-17 -t 1
[2020-01-23T10:48:06.1579765686] Installing dependencies
[2020-01-23T10:48:13.1579765693] Starting InfluxDB
[2020-01-23T10:48:13.1579765693] InfluxDB starting at :48081
[2020-01-23T10:48:17.1579765697] Executing: spark-submit --jars /Users/tarik.yilmaz/.spark-flamegraph/statsd-jvm-profiler.jar --conf spark.executor.extraJavaOptions=-javaagent:statsd-jvm-profiler.jar=server=10.118.0.200,port=48081,reporter=InfluxDBReporter,database=profiler,username=profiler,password=profiler,prefix=sparkapp,tagMapping=spark --master local[*] --executor-memory 8g example-job-0.1.jar -i file:///tmp/input -o file:///tmp/output -d 2020-01-17 -t 1 --driver-java-options -javaagent:/Users/tarik.yilmaz/.spark-flamegraph/statsd-jvm-profiler.jar=server=10.118.0.200,port=48081,reporter=InfluxDBReporter,database=profiler,username=profiler,password=profiler,prefix=sparkapp,tagMapping=spark
20/01/23 10:48:19 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
Exception in thread "main" org.apache.commons.cli.UnrecognizedOptionException: Unrecognized option: --driver-java-options
	at org.apache.commons.cli.Parser.processOption(Parser.java:363)
	at org.apache.commons.cli.Parser.parse(Parser.java:199)
	at org.apache.commons.cli.Parser.parse(Parser.java:85)
	at com.metglobal.cli.CommandLineFactory.parse(CommandLineFactory.java:40)
	at com.metglobal.App$.main(App.scala:10)
	at com.metglobal.App.main(App.scala)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.spark.deploy.JavaMainApplication.start(SparkApplication.scala:52)
	at org.apache.spark.deploy.SparkSubmit.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:849)
	at org.apache.spark.deploy.SparkSubmit.doRunMain$1(SparkSubmit.scala:167)
	at org.apache.spark.deploy.SparkSubmit.submit(SparkSubmit.scala:195)
	at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
	at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:924)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:933)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
20/01/23 10:48:19 INFO util.ShutdownHookManager: Shutdown hook called
20/01/23 10:48:19 INFO util.ShutdownHookManager: Deleting directory /private/var/folders/wb/ld7p8m8d1sj6r7cv4p_xdmxw0000gn/T/spark-8149b640-4104-4709-9a5e-7e5f06c44926
[2020-01-23T10:48:19.1579765699] Spark has exited with bad exit code (1)
[2020-01-23T10:48:19.1579765699] Collecting profiling metrics
[2020-01-23T10:48:19.1579765699] No profiling metrics were recorded!
```

What was the executed command,
```
spark-submit 
    --jars /Users/tarik.yilmaz/.spark-flamegraph/statsd-jvm-profiler.jar 
    --conf spark.executor.extraJavaOptions=-javaagent:statsd-jvm-profiler.jar=server=10.118.0.200,port=48081,reporter=InfluxDBReporter,database=profiler,username=profiler,password=profiler,prefix=sparkapp,tagMapping=spark 
    --master local[*]
    --executor-memory 8g 
    example-job-0.1.jar 
    -i file:///tmp/input 
    -o file:///tmp/output 
    -d 2020-01-17 
    -t 1 
    --driver-java-options -javaagent:/Users/tarik.yilmaz/.spark-flamegraph/statsd-jvm-profiler.jar=server=10.118.0.200,port=48081,reporter=InfluxDBReporter,database=profiler,username=profiler,password=profiler,prefix=sparkapp,tagMapping=spark
```
What should be,
```bash
spark-submit 
    --jars /Users/tarik.yilmaz/.spark-flamegraph/statsd-jvm-profiler.jar 
    --conf spark.executor.extraJavaOptions=-javaagent:statsd-jvm-profiler.jar=server=10.118.0.200,port=48081,reporter=InfluxDBReporter,database=profiler,username=profiler,password=profiler,prefix=sparkapp,tagMapping=spark
    --driver-java-options -javaagent:/Users/tarik.yilmaz/.spark-flamegraph/statsd-jvm-profiler.jar=server=10.118.0.200,port=48081,reporter=InfluxDBReporter,database=profiler,username=profiler,password=profiler,prefix=sparkapp,tagMapping=spark
    --master local[*]
    --executor-memory 8g 
    example-job-0.1.jar 
    -i file:///tmp/input 
    -o file:///tmp/output 
    -d 2020-01-17 
    -t 1
```